tests/samples/cmake-files/targetLinkLibrariesWithDynamicLookup.cmake

```diff
@@ -0,0 +1,278 @@
+
+    add_library(counter MODULE counter.c)
+    set_target_properties(counter PROPERTIES PREFIX \"\")
+  ")
+
+  if(_args_DYNAMIC_LOOKUP)
+    file(APPEND "${test_project_src_dir}/CMakeLists.txt" "
+    set_target_properties(
+      counter
+      PROPERTIES LINK_FLAGS \"-undefined dynamic_lookup\")
+    ")
+  elseif(_args_SYMBOL_DEDUPE)
+    file(APPEND "${test_project_src_dir}/CMakeLists.txt" "
+    target_link_libraries(counter number)
+    ")
+  endif()
+
+  file(APPEND "${test_project_src_dir}/CMakeLists.txt" "
+    add_executable(main main.c)
+    target_link_libraries(main number)
+    target_link_libraries(main ${CMAKE_DL_LIBS})
+  ")
+
+  file(WRITE "${test_project_src_dir}/number.c" "
+    #include <number.h>
+
+    static int _number;
+    void set_number(int number) { _number = number; }
+    int get_number() { return _number; }
+  ")
+
+  file(WRITE "${test_project_src_dir}/number.h" "
+    #ifndef _NUMBER_H
+    #define _NUMBER_H
+    extern void set_number(int);
+    extern int get_number(void);
+    #endif
+  ")
+
+  file(WRITE "${test_project_src_dir}/counter.c" "
+    #include <number.h>
+    int count() {
+      int result = get_number();
+      set_number(result + 1);
+      return result;
+    }
+  ")
+
+  file(WRITE "${test_project_src_dir}/counter.h" "
+    #ifndef _COUNTER_H
+    #define _COUNTER_H
+    extern int count(void);
+    #endif
+  ")
+
+  file(WRITE "${test_project_src_dir}/main.c" "
+    #include <stdlib.h>
+    #include <stdio.h>
+    #include <dlfcn.h>
+
+    int my_count() {
+      int result = get_number();
+      set_number(result + 1);
+      return result;
+    }
+
+    int main(int argc, char **argv) {
+      void *counter_module;
+      int (*count)(void);
+      int result;
+
+      counter_module = dlopen(\"./counter.so\", RTLD_LAZY);
+      if(!counter_module) goto error;
+
+      count = dlsym(counter_module, \"count\");
+      if(!count) goto error;
+
+      result = count()    != 0 ? 1 :
+               my_count() != 1 ? 1 :
+               my_count() != 2 ? 1 :
+               count()    != 3 ? 1 :
+               count()    != 4 ? 1 :
+               count()    != 5 ? 1 :
+               my_count() != 6 ? 1 : 0;
+
+
+      goto done;
+      error:
+        fprintf(stderr, \"Error occured:\\n    %s\\n\", dlerror());
+        result = 1;
+
+      done:
+        if(counter_module) dlclose(counter_module);
+        return result;
+    }
+  ")
+
+  set(_rpath_arg)
+  if(APPLE AND ${CMAKE_VERSION} VERSION_GREATER 2.8.11)
+    set(_rpath_arg "-DCMAKE_MACOSX_RPATH='${CMAKE_MACOSX_RPATH}'")
+  endif()
+
+  try_compile(project_compiles
+              "${test_project_bin_dir}"
+              "${test_project_src_dir}"
+              "${projectName}"
+              CMAKE_FLAGS
+                "-DCMAKE_SHARED_LINKER_FLAGS='${CMAKE_SHARED_LINKER_FLAGS}'"
+                ${_rpath_arg}
+              OUTPUT_VARIABLE compile_output)
+
+  set(project_works 1)
+  set(run_output)
+
+  if(project_compiles)
```

Comment:
We should implement the same behaviour as `try_run `.... 
See https://cmake.org/cmake/help/git-next/command/try_run.html#behavior-when-cross-compiling